### ISSUE TITLE
Add a global search bar in notification popup

### ISSUE THREAD

**Issue by xi7** [falzonv/discreet-launcher#61]
![Screenshot_2021-05-30-06-55-47](https://user-images.githubusercontent.com/67331785/120101842-8a653400-c11e-11eb-82f2-b6ca25b158eb.png)

0- based on what was used and removing your old concept before folders.
1- grid_view is folder, favorites is folder also notifications.
2- search bar
3- fit window and keyboard w/o suggestion
4- while search, always show full list apps in any scenario.
5- highlights label while typing overwrite other color

**Comment by falzonv** [comment]
Have you tried Keikai Launcher? (https://f-droid.org/en/packages/com.anpmech.launcher)
This is a lightweight search-based launcher that also has a persistent notification. From what I see, it does exactly what you described so you may like it.

If you like the empty home screen of Discreet Launcher, you could even mix both (using Discreet Launcher home screen with the notification of Keika Launcher).

---

> 0- based on what was used and removing your old concept before folders.
1- grid_view is folder, favorites is folder also notifications.

=> This looks like cancelling the current application and create a new one based only on one big folder.
Sorry, this is not what I want to do with this launcher and I do not have time to manage a new application.

> 2- search bar
> 3- fit window and keyboard w/o suggestion
> 4- while search, always show full list apps in any scenario.

=> Yes perhaps we could add a search bar...

With folders the user can organize applications (for example by theme like \"Games\", \"Tools\", ...), so I am not sure if a search tool is really useful when the applications are already organized in folders (maybe, if the user has many folders and do not remember in which one (s)he stored the application, it can be useful).
*@AeOeAa: I know you have many applications, do you think a search bar would be useful to you?*

Even if it is technically possible to put all the applications in one folder like you did, this is not the way folders are supposed to be used in Discreet Launcher (it is why I suggested you to check Keikai Launcher which is more in the vision you described).

> 5- highlights label while typing overwrite other color

=> Not sure if this part will be possible to be honest.

---

I will keep this ticket for the search bar idea.

But replacing what you call "my old concept" (favorites, grid view, etc. ; in two words, "Discreet Launcher") with a folder will not be done. This is not the development direction I want to give to this launcher.

**Comment by xi7** [comment]
**about the keika, completely useless the notification bar, better press home and go to home screen,**
**you can just add like the leika primitive bar search on grid view... but is ugly and not discreet.**

------
what has been implemented is already more efficient than the others. (v.3.6.0 with notifications / folders) 
but if you look at the photo, nothing has changed with search bar, remains discreet.

**Comment by falzonv** [comment]
Ok, so in fact what you need is an "all apps search bar" in the notification popup.
Good idea indeed, probably not so easy to implement but I will have a look ;-)

**Comment by xi7** [comment]
![Screenshot_2021-05-30-09-57-42](https://user-images.githubusercontent.com/67331785/120307714-af39e280-c2a9-11eb-94ab-b3c07cc57ed5.png)

u can use this one very discreet like a shortcut search house. hide/unhide option the main_search.

**and now what you think ?**

**Comment by xi7** [comment]
illustrative pic favorites, but the example is to be used in all scenarios.

0- the first is always main_search house. (no need name)
1- enabled by default and also key enable/disable settings area.
2- keyboard W/O suggestion

**Comment by xi7** [comment]
the same idea can be used here (#58 house menu settings) while hide.

**Comment by falzonv** [comment]
Great idea, the search house looks very nice!
Thank you for the suggestion :-)

---

For the menu button in #59, I was thinking to directly reuse the Discreet Launcher app icon :
- Short click would display the menu
- Long click would display the dialog with "Open" (access menu) and "Settings" (access Discreet Launcher app settings)

I have not started yet to work on it so probably I will try both ideas to see which one looks better.

**Comment by xi7** [comment]
intuitive follow the house and would keep a pattern.

if u like change the color house, just for fast viewing

**Comment by falzonv** [comment]
Hello,

I am pleased to inform you that the search has been implemented in new release v4.1.0!
This release has been published a few minutes ago.

Here are the usual availability delays for each location:
- 4 to 8 days on F-Droid
- 0 to 2 days on Google Play
- Already available as direct APK on GitHub (https://github.com/falzonv/discreet-launcher/releases)

*Remarks:
I will leave this ticket opened until the release is effectively published on F-Droid.
You can use the export/import feature to switch from one version to another.*

Best regards.

**Comment by xi7** [comment]
![Screenshot_2021-06-19-07-42-58](https://user-images.githubusercontent.com/67331785/122640286-273f3f80-d0d5-11eb-802e-a551cb3ffdd5.png)

android 5 i have a black target folder

**Comment by xi7** [comment]
working very good, next release possible add on/off favorites home ?

**Comment by xi7** [comment]
![Screenshot_2021-06-19-08-13-53](https://user-images.githubusercontent.com/67331785/122640544-8782b100-d0d6-11eb-8c94-7f486123968a.png)

add after press enter, auto open the first app, because when press enter have a bug area line.

**Comment by falzonv** [comment]
Thank you for your feedback! :-)

> next release possible add on/off favorites home ?

=> You mean, add an option to always hide the favorites panel? (that is, the reverse option of #70?)
Perhaps, maybe it could be a global setting "Favorites panel visibility" with options "Normal" / "Always visible" / "Always hidden"...

> add after press enter, auto open the first app, because when press enter have a bug area line

=> Ok, not sure if I can attach an action to the Enter button but I will have a look.

**Comment by xi7** [comment]
> => You mean, add an option to always hide the favorites panel? (that is, the reverse option of #70?)
> Perhaps, maybe it could be a global setting "Favorites panel visibility" with options "Normal" / "Always visible" / "Always hidden"...

not this, add the house search bar favorites home too, list area select favorites.

**Comment by falzonv** [comment]
Oh ok, this is already possible using the favorites selector.
For technical reasons I cannot have the Search first in the selection list, but it is at its place in the alphabet (look for "Search").

**Comment by xi7** [comment]
ok i see, tnks. perfect ! 4.1.0 working

**Comment by falzonv** [comment]
Hello,

New release v4.2.0 is now published on GitHub!

It integrates:

> add after press enter, auto open the first app, because when press enter have a bug area line

=> Not exactly like this but very similar: when testing, I found it was more intuitive to auto open only when there is 1 search result remaining (like in your screenshot). If there are 2 or 3 results remaining, the user may not want to launch the first one so (s)he can either click on the desired icon, or add some characters to filter on the correct app before pressing "Enter".

> The workaround mentioned in ticket #67 for the keyboard in the landscape search

=> It should solve the issues with most keyboards, as long as they support the "flagNoExtractUi" Android attribute.

---

Since the search is now implemented and v4.1.0 is published since Thursday, I will close this ticket.
Feel free to add comments if needed or wished.
